util/awsservice/constant.go

```diff
@@ -56,12 +58,23 @@ var (
 
 func init() {
 	ctx = context.Background()
-	var err error
-	awsCfg, err := config.LoadDefaultConfig(ctx, config.WithRegion("us-west-2"))
+
+	err := ConfigureAWSClients("us-west-2")
+	if err != nil {
+		fmt.Println("There was an error trying to configure the AWS clients: ", err)
+	}
+}
+
+// ConfigureAWSClients configures the AWS clients using a set region.
+func ConfigureAWSClients(region string) error {
+	mu.Lock()
+	defer mu.Unlock()
```

Comment:
Why do we need a lock?

Reply:
We need a lock of some kind because when we call `ConfigureAWSClients` inidividually, we do after we call the `init` function implicitly through another package. The lock prevents both the `init` and `ConfigureAWSClients` from changing the client values at the same time.